### ISSUE TITLE
terraform plan requires write access to s3

### ISSUE THREAD

**Issue by grayaii** [hashicorp/terraform#5815]
Our TF state files are in s3.
We have a nightly jenkins job that runs "terraform plan" to see if anything has changed, and if so, we get alerted.

Unfortunately "terraform plan" requires s3 WRITE access. 

We don't want terraform plan WRITE access to the state files because we don't want them inadvertently modifying the state files.

```
  export TF_VAR_2nd_octet=55
  export TF_VAR_region=us-west-2
  export TF_VAR_owner=foo@foo.com
  export TF_VAR_availability_zones=us-west-2a,us-west-2b,us-west-2c
  export TF_VAR_aws_access_key=****
  export TF_VAR_aws_secret_key=****
  export TF_VAR_owner=alexgray
  export TF_VAR_aws_account_id=****
  export TF_VAR_environment=alex-abc
  export TF_VAR_backend=s3
  export TF_VAR_region=us-west-2
  export TF_VAR_ssh_key=****
  export TF_VAR_action=plan

  terraform remote config -backend-config="name=alex-abc/vpc" -backend-config="bucket=298394155355-tf-remote-state" -backend-config="key=alex-abc/vpc" -backend-config="region=us-west-2" -backend=S3

  terraform get
  # THE BELOW COMMAND FAILS WITH ACCESS DENIED TO S3:
  terraform plan -input=false
```

Does "terraform plan" REALLY need s3 write access?

**Comment by jen20** [comment]
Hi - right now this is the case - though you could plan by copying the state locally and running it from there. There is a roadmap item likely to hit in 0.7 which prevents creating a new state version on plan.

**Comment by grayaii** [comment]
Awesome! Thanks!  For now, ```-refresh=false``` works fine for our purposes.

**Comment by phoolish** [comment]
I believe this is solved with 0.7.0

```
The terraform plan command no longer persists state. This makes the command much safer to run, since it is now side-effect free. The refresh and apply commands still persist state to local and remote storage. Any automation that assumes that terraform plan persists state will need to be reworked to explicitly call terraform refresh to get the equivalent side-effect. (The terraform plan command no longer has the -state-out or -backup flags due to this change.)
```

**Comment by mitchellh** [comment]
It is, thanks @phoolish
